lib/doh.c

```diff
@@ -182,6 +186,33 @@ doh_write_cb(const void *contents, size_t size, size_t nmemb, void *userp)
   return realsize;
 }
 
+#ifdef USE_HTTPSRR
+static void local_print_buf(struct Curl_easy *data,
+                            unsigned char *buf, size_t len)
+{
+  size_t j;
+  char string_buffer[2048], *ptr;
+
+  ptr = string_buffer;
+  if(len > 1000) {
+    infof(data, "\t:buffer too long %ld", len);
+    return;
+  }
+  for(j = 0; j < len; j ++) {
+    *ptr++ = BIN2HEXDIGIT(buf[j] / 16);
+    *ptr++ = BIN2HEXDIGIT(buf[j] % 16);
+    if((ptr - string_buffer) > 2000) {
```

Comment:
Can this even happen since you already check `len` before this loop starts?

Reply:
yep, could be cleaned up, but is there an existing function to use instead of adding a new ``local_print_buf()`` ? If so, I'd use that and get rid of this entirely.

Reply:
libcurl does not have any generic "hex encode" function, but I agree it would make sense. I'll make one in a separate PR! See  #11990

Reply:
Now making use of the #11990 function.